airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -91,9 +93,16 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
         Override this method to define how a response is parsed.
         :return an iterable containing each record in the response
         """
-        records = response.json().get(self.name)
-        for record in records:
-            yield record
+        try:
+            datasets = response.json().get(self.name)
+            for dataset in datasets:
+                yield dataset
+        except TypeError as e:
+            raise AirbyteTracedException(
+                            internal_message=str(e),
+                            failure_type=FailureType.config_error,
+                            message="Provided crendentials do not give access to any datasets or project has no datasets",
```

Comment:
I like that this is actionable. Should we give more context here though? Something like `Could not retrieve any projects or datasets associated to the projects. Provided credentials do not give access to any datasets or project has no datasets` 

nit: `credentials` instead of `crendentials`